app/scripts/views/complete_sign_up.js

```diff
@@ -60,6 +64,17 @@ function (Cocktail, FormView, BaseView, CompleteSignUpTemplate,
       return self.fxaClient.verifyCode(uid, code)
           .then(function () {
             self.logScreenEvent('verification.success');
+            var account = self.getAccount();
+
+            if (account.get('needsOptedInToMarketingEmail')) {
```

Comment:
If the user is verifying in a different browser than they signed up with, this will be undefined. We may have to use the resume token if we want to reliably honor the opt-in.